### ISSUE TITLE
User profile data retrieval

### ISSUE THREAD

**Issue by george-i** [firebase/firebase-login-demo-android#7]
Hi,

I have a few questions/remarks and I don't know if this is expected. 

- twitter: user clicks the login with Twitter button.  The "connect Twitter account with app" webview is shown. The user clicks "cancel" and the app crashes.

- facebook: the user selects facebook as login method. Later, the user logs out from app using unauth method. When the user click login with facebook again, a webview is shown where it says that the user is already connected to the app and I can't retrieve user profile. 
- twitter again: although the user has selected twitter as login method, if he logs out from the app using unauth method and wants to login again with Twitter, a webview is shown and the user is asked again if he wants to connect his profile with the app.

Basically, in my app, this is what happens:
The user is in home activity. 
He clicks on login button and the login activity is opened. 
The user selects one of the social networks to login.
Once logged in, the user is sent back to home activity. 
On the home activity, the user clicks logout. 
Later, he can open again the login activity.

**Comment by george-i** [comment]
Any ideas?

**Comment by oalami** [comment]
Hi George - I'm not sure what you're asking here. Are you referring to your app or the sample? Login is delegated to the respective provider (facebook or twitter) to handle, you then use the returned auth token to auth to Firebase as that user.

**Comment by george-i** [comment]
Let's take them one by one.
> twitter: user clicks the login with Twitter button. The "connect Twitter account with app" webview is shown. The user clicks "cancel" and the app crashes.

Can you verify this?

**Comment by markbratanov** [comment]
@george-i I can verify this. I came here exactly to see if there was an updated solution, haha.

I'm currently looking into using Twitter's Fabric SDK with Firebase Auth. I'll let you guys know of any advancements.

**Comment by ardock** [comment]
![screenshot-311](https://cloud.githubusercontent.com/assets/1142360/8458629/70fd5dc4-2018-11e5-9e05-b3d0ddfbf348.png)

**Comment by ardock** [comment]
@puf If the user cancels Twitter login, the demo tries to retrieve a token from a null object and the app crashes.

If the response is `Oauth://cb?denied=...` (Image lines 76 and 77), you need to do something like this:

    if (url.startsWith("oauth://cb")) {
        if (url.contains("oauth_verifier")) {
            ...
        }
        else if (url.contains("denied")) {
            ...
        }
    }

I thought the image and the pull request was enough but you ignored this issue and answered other, so I'm trying to explain it better. 

Really I don't use neither Twitter nor Firebase but the Google I/O announcement about offline support was interesting.

Good job, but please update the sample, all the Google sign-in code is duplicated etc. I understand that a demo needs to be simple but currently it's not a Google style sample. Thanks.

**Comment by puf** [comment]
@ardock Thanks for the follow up and the PR. Sorry about not getting back to you sooner. My main problem with the issue is that I've been unable to reproduce it. Of course that changed when I tried again today. I'll have a look at your PR in the coming days. But ping me if I forget again!

**Comment by ardock** [comment]
@puf Ok hehe, np, thanks for answer.
